Late escape Post blocks

getdave opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

This is not a security problem. This PR simply moves [escaping of all PHP output to be as "late" as possible](https://developer.wordpress.org/themes/theme-security/data-sanitization-escaping/#:~:text=esc_textarea(%20%24text%20)%3B%20%3F%3E%3C/textarea%3E-,Tip%3A,Output%20escaping%20should%20occur%20as%20late%20as%20possible.,-Top%20%E2%86%91). This means we avoid escaping variables until they are output in the HTML markup.

This is a WP Core best practice.

## How has this been tested?
- add both blocks
- check functionality is "as was"
- check all tests pass


## Screenshots <!-- if applicable -->

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->
- [ ] I've updated related schemas if appropriate. <!-- Reference: https://github.com/WordPress/gutenberg/tree/trunk/schemas -->

getdave committed:
Post date initial pass

getdave committed:
Post excerpt initial pass

getdave committed:
Feature image block first pass

getdave committed:
Post Navigation link first pass

getdave committed:
Post Template first pass

getdave committed:
Post terms first pass

getdave committed:
First pass Post title

aristath commented on packages/block-library/src/post-featured-image/index.php:
```diff
@@ -35,15 +35,15 @@ function render_block_core_post_featured_image( $attributes, $content, $block )
 			$image_styles .= "object-fit:{$attributes['scale']};";
 		}
-		$featured_image = str_replace( 'src=', "style='$image_styles' src=", $featured_image );
+		$featured_image = str_replace( 'src=', 'style="' . esc_attr( $image_styles ) . '" src=', $featured_image );
```
Is there a reason why the `"` was changed to `'` in `src`? 🤔

getdave replied:
I am getting WordPress Coding Standards warnings about using double quotes for strings. That said I know both are deemed acceptable. 

But as we're not interpolating variables anymore single quotes are probably best suited here.

aristath commented on packages/block-library/src/post-template/index.php:
```diff
@@ -62,7 +62,7 @@ function render_block_core_post_template( $attributes, $content, $block ) {
 			)
 		)->render( array( 'dynamic' => false ) );
 		$post_classes  = esc_attr( implode( ' ', get_post_class( 'wp-block-post' ) ) );
```
Since we're moving escaping, it should be removed from this line, otherwise it gets double-escaped.
```suggestion
		$post_classes  = implode( ' ', get_post_class( 'wp-block-post' ) );
```

getdave replied:
Yeh let's avoid double escaping! 👍

getdave committed:
Avoid double escaping

getdave commented:
Thank you for your patience @aristath 🙇

getdave committed:
Remove redunant escaping

`get_block_wrapper_attributes` will escape for us. This is as per feedback on other PRs involving late escaping.

scruffian commented:
This creates a problem when the post date is a link:
<img width="508" alt="Screenshot 2022-01-14 at 17 14 46" src="https://user-images.githubusercontent.com/275961/149556907-3da6bde1-9eda-4eb9-9616-cfb888bea135.png">

getdave commented:
@scruffian Let me fix that now.
